vme/src/db.cpp

```diff
@@ -856,57 +856,52 @@ unit_data *read_unit_string(CByteBuffer *pBuf, int type, int len, const char *wh
             /* fallthru */
 
         case UNIT_ST_PC:
-            CHAR_EXP(u) = pBuf->ReadS32(&g_nCorrupt);
-            CHAR_FLAGS(u) = pBuf->ReadU32(&g_nCorrupt);
+        {
+            auto *character = dynamic_cast<char_data *>(u);
+            character->points.readPlayerExperienceFrom(*pBuf, g_nCorrupt);
+            character->points.readCharacterFlagsFrom(*pBuf, g_nCorrupt);
 
-            CHAR_MANA(u) = pBuf->ReadS16(&g_nCorrupt);
-            CHAR_ENDURANCE(u) = pBuf->ReadS16(&g_nCorrupt);
+            character->points.readManaFrom(*pBuf, g_nCorrupt);
```

Comment:
Cool - I was doing that for ease of review - I prefer it done from 1 function in the class too. Although can we pick a longer more descriptive name than 'bread' :-)  I always think of a loaf of bread first

I like the `From` in the name - so that way we can create overloads that read from JSON, text, binary, BSON or whatever. 
ReadCharacterPointsFrom(JSON &)
ReadCharacterPointsFrom(CByteBuffer &)
ReadCharacterPointsFrom(std::istream &)
or maybe
LoadCharacterPointsFrom
Or how about the whole hog c++ paradigm with operator>>()

Reply:
I would leave something like >> until the very end when we're through conversion. Right now it'd end up as a hodge podge IMO and cause code change in too many other places too.

I think ReadCharacterPointsFrom() is nice. But then we should probably rename the other ones too? What about bread_unit_string() for example, do you want to ReadUnitFrom(&CByteBuffer) ? That would be nice. I guess it's a timing question.